### PR TITLE
fallback name proposal and name proposal save bypass

### PR DESCRIPTION
**Pull request by ix0rai**
implements #239 

- adds a `bypassValidation()` boolean in `NameProposalService`
  - this boolean allows that service to ignore the regular checks performed on proposed mappings (validating that they either remove a mapping or provide a proposed mapping)
  - this is essential for part of https://github.com/QuiltMC/quilt-mappings/issues/660, since the mappings proposed for classes to apply moj packages have to be `TokenType.DEOBFUSCATED`
- adds a `fallback()` boolean to `NameProposalService`
  - fallback names will not count towards stats by default
  - fallback names will have a different colour in the UI
  - fallback names do not have their own token types, and are expected to be less reliably high-quality names than normal proposed mappings

### CODE REVIEW

File: enigma/src/test/java/org/quiltmc/enigma/name_proposal/TestFallbackNameProposal.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+						]
+					}
+				}""");
+
+		try {
+			EnigmaProfile profile = EnigmaProfile.parse(r);
+			Enigma enigma = Enigma.builder().setProfile(profile).setPlugins(List.of(new BuiltinPlugin(), new TestPlugin())).build();
+			project = enigma.openJar(JAR, new ClasspathClassProvider(), ProgressListener.createEmpty());
+		} catch (Exception e) {
+			Logger.error(e, "Failed to open jar!");
```
Comment (supersaiyansubtlety) [inline enigma/src/test/java/org/quiltmc/enigma/name_proposal/TestFallbackNameProposal.java:69]: Should this re-throw?

File: enigma/src/test/java/org/quiltmc/enigma/name_proposal/TestFallbackNameProposal.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	private static EnigmaProject project;
+
+	@BeforeAll
+	public static void setupEnigma() {
+		Reader r = new StringReader("""
+				{
+					"services": {
+						"name_proposal": [
+							{
+								"id": "test:name_all_fields_slay"
```
Comment (IotaBread) [inline enigma/src/test/java/org/quiltmc/enigma/name_proposal/TestFallbackNameProposal.java:54]: i love  these names

**Review by IotaBread — APPROVED** [review]
Looks great!

File: enigma/src/test/java/org/quiltmc/enigma/name_proposal/TestFallbackNameProposal.java (hunk 2 above)
Comment (ix0rai) [reply enigma/src/test/java/org/quiltmc/enigma/name_proposal/TestFallbackNameProposal.java:54]: i never change <3

**Pull request merged**